src/services/blocks/BlocksService.ts

```diff
@@ -45,11 +44,21 @@ export class BlocksService extends AbstractService {
 	): Promise<IBlock> {
 		const { api } = this;
 
-		const [{ block }, events, validators] = await Promise.all([
-			api.rpc.chain.getBlock(hash),
-			this.fetchEvents(api, hash),
-			api.query.session.validators.at(hash),
-		]);
+		let block;
+		let events;
+		let author;
+		if (typeof api.query?.session?.validators?.at === 'function') {
```

Comment:
This looks super esoteric to me, why do we check for the existence of a deeply nested function, and then not call it or do anything with it. Any sort of implicit connection between this api, and the api we end up calling, could at least use a comment.